File: DAY4/loops_acc_implicit.c
#include<stdio.h>
#include<stdlib.h>

int main()
{

        int n=1000000000;

        int * a = ( int * )  malloc( n * sizeof(int) );
        int * b = ( int * )  malloc( n * sizeof(int) );
        int * c = ( int * )  malloc( n * sizeof(int) );

#pragma acc parallel loop
        for(int i=0; i<n; i++){
                a[i]=3;
                b[i]=4;
        }

#pragma acc parallel loop
        for(int j=0; j<n; j++){
                a[j] = 2*a[j];
                b[j] = b[j]+1;
        }

#pragma acc parallel loop
        for(int k=0; k<n; k++){
                c[k] = a[k]+b[k];
        }
	
      
        for(int l=0; l<10; l++){
                printf("c[%d]=%d\n",l, c[l]);
           }
        printf("\n");

        return 0;
}

File: DAY4/loops_acc_implicit.f90
program loops
      use openacc
      implicit none
      
      integer,allocatable :: a(:),b(:),c(:) 
      integer(kind=8)     :: n
      integer             :: i,j,k
      n=10_8**10

      allocate(a(n))
      allocate(b(n))
      allocate(c(n))
!$acc parallel loop
     do i=1,n
        a(i)=3
        b(i)=4
     end do

!$acc parallel loop
     do j=1,n
        a(j) = 2*a(j)
        b(j) = b(j) +1      
     end do 

!$acc parallel loop     
     do k=1,n
        c(k) = a(k) +b(k)
     end do   

     print *, c(1:10)
end program  
